### PR TITLE
VZ-7558 Improvisations of Helidonworkload Code Coverage 

### PR DESCRIPTION
**Pull request by absBitm**

### CODE REVIEW

**Comment by anajosep** [comment]
Please provide some description and address the 4 code smells reported by sonarcloud.

File: application-operator/controllers/helidonworkload/helidonworkload_controller_test.go [hunk 1]
```diff
@@ -980,3 +982,71 @@ func TestReconcileFailed(t *testing.T) {
 	reconcileFailedCounterAfter := testutil.ToFloat64(reconcileerrorCounterObject.Get())
 	assert.Equal(reconcileFailedCounterBefore, reconcileFailedCounterAfter-1)
 }
+
+func TestAddMetrics(t *testing.T) {
```
Comment (adiforluls) [inline application-operator/controllers/helidonworkload/helidonworkload_controller_test.go:986]: Add func comment like this
```
TestXYZ tests the func XYZ for the following use case
GIVEN ....
WHEN ....
THEN ....
```

Comment (absBitm) [reply application-operator/controllers/helidonworkload/helidonworkload_controller_test.go:987]: done

**Review by adiforluls — APPROVED** [review]

**Review by anajosep — APPROVED** [review]
LGTM

**Pull request merged**